### DISCUSSION TITLE
Error on Build

### DISCUSSION THREAD

**Discussion by lexiekark** [perspective-dev/perspective#1973] [Q&A] [answered]
Hello! I am encountering this error when I try to build all of perspective.

`[ 51%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/csv/parser.cc.o
[ 51%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/csv/reader.cc.o
[ 51%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/csv/writer.cc.o
[ 51%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/ipc/dictionary.cc.o
[ 53%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/ipc/feather.cc.o
[ 53%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/ipc/message.cc.o
[ 53%] Building CXX object arrow-build/CMakeFiles/arrow.dir/src/arrow/ipc/metadata_internal.cc.o
In file included from /Users/lexiekarkazis/perspective/cpp/perspective/dist/release/arrow-src/cpp/src/arrow/ipc/feather.cc:36:
In file included from /Users/lexiekarkazis/perspective/cpp/perspective/dist/release/arrow-src/cpp/src/arrow/ipc/metadata_internal.h:41:
In file included from /Users/lexiekarkazis/perspective/cpp/perspective/dist/release/arrow-src/cpp/src/generated/Message_generated.h:9:
/Users/lexiekarkazis/perspective/cpp/perspective/dist/release/arrow-src/cpp/src/generated/Schema_generated.h:775:12: error: no matching member function for call to 'VerifyField'
           VerifyField<int32_t>(verifier, VT_LISTSIZE) &&
           ^~~~~~~~~~~~~~~~~~~~
/usr/local/include/flatbuffers/table.h:115:8: note: candidate function template not viable: requires 3 arguments, but 2 were provided
  bool VerifyField(const Verifier &verifier, voffset_t field,
       ^`

I get a bunch of this same error before fully failing and giving me this final message:
`make[2]: *** [arrow-build/CMakeFiles/arrow.dir/src/arrow/ipc/metadata_internal.cc.o] Error 1
make[1]: *** [arrow-build/CMakeFiles/arrow.dir/all] Error 2
make: *** [all] Error 2
emmake: error: 'make -j4' failed (returned 2)
Error: Command failed: emmake make -j4
    at checkExecSyncError (child_process.js:790:11)
    at execSync (child_process.js:863:15)
    at Object.<anonymous> (/Users/lexiekarkazis/perspective/cpp/perspective/build.js:20:5)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 10405,
  stdout: null,
  stderr: null
}
Command failed: cd /Users/lexiekarkazis/perspective/.emsdk && . ./emsdk_env.sh >/dev/null 2>&1 && cd /Users/lexiekarkazis/perspective/cpp/perspective && node ./build.js
error Command failed with exit code 1.`

I have brew installed boost, cmake, and flatbuffers.  I am also using Node 18.9.0.

Thank you in advance for any assistance!

**Comment by timkpaine** [comment, marked as answer]
Uninstall flatbuffers from brew, there are compatibility issues we're still looking into.
